mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -223,6 +235,49 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
   return success();
 }
 
+LogicalResult GemmRewritePattern::computeGridSize(ConversionPatternRewriter &rw,
+                                                  GemmOp op, Value a,
+                                                  Value b) const {
+  GemmFeatures features = op.getGemmFeatures();
+  Attribute params = op.getParams().value();
+
+  auto aShape = a.getType().cast<MemRefType>().getShape();
+  auto bShape = b.getType().cast<MemRefType>().getShape();
+
+  int64_t G = aShape[0];
+  int64_t M = aShape[2];
+  int64_t N = bShape[2];
+
+  auto obtainGridSize = [&](auto tuningParams) {
+    return (M / tuningParams.getMPerBlock()) *
+           (N / tuningParams.getNPerBlock()) * G;
+  };
+
+  int64_t gridSize{0};
+  if (isAccel(features)) {
+    auto tuningParams = params.cast<RockAccelTuningParamAttrInterface>();
+    gridSize = obtainGridSize(tuningParams);
+
+    if (auto bwdOp = llvm::dyn_cast<Conv2DBwdWeightOp>(*op)) {
```

Comment:
Wait, how isn't trivially always false? `op` is always a `GemmOp`

I think at this point in the lowering the blocks have already been added to the `G` dimension so you don't need this.

Reply:
Hi, yes. It makes total sense. I don't know why I forgot about it. Thanks for noticing!

Done.